### DISCUSSION TITLE
Difference between these two way of autostart on `fish` spawned

### DISCUSSION THREAD

**Discussion by leonasdev** [zellij-org/zellij#2515] [Q&A] [answered]
In the documentation, it mention that you can use:
```fish
if set -q ZELLIJ
else
  zellij
end
```
to autostart zellij when fish is spawned, but it instantly continue with:
```fish
if status is-interactive
    ...
    eval (zellij setup --generate-auto-start fish | string collect)
end
```

So idk are these two way autostart differently or just documentation is not updated yet?

sorry if it is a dumb question, not quite familiar with either zellij or fish script.

**Comment by Lockszmith-GH** [comment]
The seconds option is what you should use. I'm guessing that the docs aren't updated.

**Reply by leonasdev** [reply]
But it seems both approach behavior the same?

**Reply by Lockszmith-GH** [reply, marked as answer]
As long as you are in an interactive session. But if you run a non-interactive session (for example `zsh -c "printf 'hello world\n'"`) BEFORE you've started zellij, in the first option, zellij would start and mess things up.
